server.go

```diff
@@ -81,22 +84,97 @@ func (s *Server) ListenAndServe(addr string) error {
 // but still want to be able to handle any errors opening connection.
 // Serve returns when Shutdown is called or connection is closed.
 func (s *Server) Serve(conn *net.UDPConn) {
+	defer conn.Close()
+	laddr := conn.LocalAddr()
+	var conn4 *ipv4.PacketConn
+	var conn6 *ipv6.PacketConn
+	host, _, err := net.SplitHostPort(laddr.String())
 	s.conn = conn
+	if err != nil {
+		panic(err.Error())
+	}
+	// Having seperate control paths for IP4 and IP6 is annoying,
+	// but necessary at this point.
+	if addr := net.ParseIP(host); addr == nil {
+		panic("Failed to parse listen address")
+	} else if addr.To4() != nil {
+		conn4 = ipv4.NewPacketConn(conn)
+		defer conn4.Close()
```

Comment:
Eh, probably.  I just tend to blindly Close() everything I open as a matter of habit.   I will look at the code again to see what calling Close() on the PacketConn does.